src/engine/server/server.cpp

```diff
@@ -4358,6 +4427,130 @@ void CServer::RemoveAccusations(int ClientID)
 	m_aClients[ClientID].m_Accusation.m_Num = 0;
 }
 
+void CServer::AddMapVote(int From, const char* pCommand, const char* pReason, const char* pDesc)
+{
+	NETADDR FromAddr = *m_NetServer.ClientAddr(From);
+	int Index = -1;
+	for (int i=0; i<m_MapVotesCounter; i++)
+	{
+		if(str_comp_nocase(m_MapVotes[i].m_pCommand, pCommand) == 0)
+		{
+			Index = i;
+			break;
+
+		}
+	}
+	if (Index < 0)
+	{		
+		Index = m_MapVotesCounter;
+		m_MapVotes[Index].m_pCommand = new char[VOTE_CMD_LENGTH];
+		str_copy(const_cast<char*>(m_MapVotes[Index].m_pCommand), pCommand, VOTE_CMD_LENGTH);
+		m_MapVotes[Index].m_pAddresses = new NETADDR[MAX_MAPVOTEADDRESSES];
+		m_MapVotes[Index].m_pAddresses[0] = FromAddr;
+		m_MapVotes[Index].m_Num = 1;
+		m_MapVotes[Index].m_pReason = new char[VOTE_REASON_LENGTH];
+		str_copy(const_cast<char*>(m_MapVotes[Index].m_pReason), pReason, VOTE_REASON_LENGTH);
+		m_MapVotes[Index].m_pDesc = new char[VOTE_DESC_LENGTH];
+		str_copy(const_cast<char*>(m_MapVotes[Index].m_pDesc), pDesc, VOTE_DESC_LENGTH);
+		m_MapVotesCounter++;
+	}
+	else 
+	{
+		if (str_comp_nocase(m_MapVotes[Index].m_pReason, "No reason given") == 0)
+			str_copy(const_cast<char*>(m_MapVotes[Index].m_pReason), pReason, VOTE_REASON_LENGTH);
+
+		for(int i=0; i<m_MapVotes[Index].m_Num; i++)
+		{
+			if(net_addr_comp(&m_MapVotes[Index].m_pAddresses[i], &FromAddr) == 0)
+			{
+				if(m_pGameServer)
+					m_pGameServer->SendChatTarget_Localization(From, CHATCATEGORY_DEFAULT, _("You have already voted to change this map"), NULL);
+				return;
+			}
+		}
+
+		m_MapVotes[Index].m_pAddresses[m_MapVotes[Index].m_Num] = FromAddr;
```

Comment:
could you please add some comments to the code about what is happening here? My parser failed on this line

Reply:
I will add some comments. Your parser failed? Does that mean it dident compile for you?